src/lint_errors.cc

```diff
@@ -15,88 +15,92 @@
 //
 #include "src/lint_errors.h"
 
-#include <vector>
-#include <iostream>
 #include <algorithm>
+#include <cstdio>
+#include <iostream>
+#include <string>
+#include <utility>
+#include <vector>
 
-#include "absl/strings/string_view.h"
 #include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
 #include "zetasql/base/status.h"
+#include "zetasql/base/status_macros.h"
+#include "zetasql/base/statusor.h"
+#include "zetasql/public/parse_helpers.h"
+#include "zetasql/public/parse_location.h"
 
 namespace zetasql::linter {
 
-absl::string_view LintError::GetErrorMessage() {
-    // Refers to the error type
-    switch ( type_ ) {
-        case kLineLimit:
-            return "Lines should be <= 100 characters long";
-        case kParseFailed:
-            return "ZetaSQL parser failed";
-        case kSemicolon:
-            return "Each statement should end with a consequtive selicolon ';'";
-        case kUppercase:
-            return "All keywords should be consist of uppercase letters";
-        case kCommentStyle:
-            return "Either '//' or '--' should be used to specify a comment";
-        case kAlias:
-            return "Always use AS keyword for referencing aliases";
-        break;
-    default:
-        break;
-    }
-    return "";
+// Copy constructor for Lint Error
+LintError::LintError(const LintError &result) {
+  type_ = result.type_;
+  filename_ = result.filename_;
+  line_ = result.line_;
+  column_ = result.column_;
+  message_ = result.message_;
+}
+
+absl::string_view LintError::GetErrorMessage() { return message_; }
+
+std::string LintError::ConstructPositionMessage() {
+  return absl::StrCat("In line ", line_, ", column ", column_, ": ");
 }
 
 void LintError::PrintError() {
-    if ( filename_ == "" ) {
-        std::cout << "In line " << line_ << ": "
-        << GetErrorMessage() << std::endl;
-    } else {
-        std::cout << filename_ << ":" << line_ << ": "
-        << GetErrorMessage() << std::endl;
-    }
+  if (filename_ == "") {
+    std::cout << ConstructPositionMessage() << GetErrorMessage() << std::endl;
+  } else {
+    std::cout << filename_ << ":" << ConstructPositionMessage()
+              << GetErrorMessage() << std::endl;
+  }
 }
 
 void LinterResult::PrintResult() {
-    // Need to sort according to increasing line number
-    sort(errors_.begin(), errors_.end(),
-        [&](const LintError &a, const LintError &b){
-        return a.getLineNumber() < b.getLineNumber();
-    });
-    for ( LintError error : errors_ )
-        error.PrintError();
+  // Need to sort according to increasing line number
+  sort(errors_.begin(), errors_.end(),
+       [&](const LintError &a, const LintError &b) {
+         return a.getLineNumber() < b.getLineNumber();
+       });
+  for (LintError error : errors_) error.PrintError();
+
+  for (absl::Status status : status_) std::cerr << status << std::endl;
+
+  std::cout << "Linter results are printed" << std::endl;
```

Comment:
Is it some debugging string that you forgot to remove?

Reply:
I think it is good to inform user that linter is finished, otherwise there would be no output when there is no lint error which is confusing for me. Though, I will erase either here or the one in 'RunChecks'. Currently erased the one in "RunChecks" since it is not a place to output results right now.

Reply:
Good idea to print some message if there were no errors found, like "No errors found".